aggregator/src/aggregator/batch_creator.rs

```diff
@@ -113,17 +109,18 @@ where
     where
         C: Clock,
     {
-        let time_bucket_start_opt = self
-            .properties
-            .task_batch_time_window_size
-            .map(|batch_time_window_size| {
-                report
-                    .client_timestamp()
-                    .to_batch_interval_start(&TimePrecision::from_seconds(
-                        batch_time_window_size.as_seconds(),
-                    ))
-            })
-            .transpose()?;
+        let time_bucket_start_opt =
+            self.properties
+                .task_batch_time_window_size
+                .map(|batch_time_window_size| {
+                    // While everything is in units of the time precision, we still have to
+                    // bucket things by the batch_time_window_size.
+                    let batch_window_units = batch_time_window_size.as_time_precision_units();
+                    Time::from_time_precision_units(
+                        (report.client_timestamp().as_time_precision_units() / batch_window_units)
+                            * batch_window_units,
```

Comment:
surely `/ batch_window_units * batch_window_units` cancels out? Then `Time::from_time_precision_units(report.client_timestamp().as_time_precision_units()) is just `report.client_timestamp()`?

Reply:
Yeah, that's the issue exactly, the possibility for them being a factor of each other. I'm going to rework that in https://github.com/divviup/janus/issues/4225.